[css-content] AX: Add support for alt text syntax

nt1m opened the pull request:
#### 1b3e497426196dd3289ad58811bdb12cbaa29379
<pre>
[css-content] AX: Add support for alt text syntax
<a href="https://bugs.webkit.org/show_bug.cgi?id=159022">https://bugs.webkit.org/show_bug.cgi?id=159022</a>
<a href="https://rdar.apple.com/26942023">rdar://26942023</a>

Reviewed by Darin Adler.

This adds support for the `content: &quot;foo&quot; / &quot;alt-text&quot;;` syntax. Multiple strings / attr() functions can also be chained.

Counters are not supported by this PR. Other non-text types are intentionally not supported as alt-text per-spec.

Spec: <a href="https://drafts.csswg.org/css-content-3/#alt">https://drafts.csswg.org/css-content-3/#alt</a>

* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content-expected.txt:
* Source/WebCore/css/CSSValuePair.cpp:
(WebCore::CSSValuePair::CSSValuePair):
(WebCore::CSSValuePair::create):
(WebCore::CSSValuePair::createSlashSeparated):
(WebCore::CSSValuePair::createNoncoalescing):
* Source/WebCore/css/CSSValuePair.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeContent):
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueContent):

Canonical link: <a href="https://commits.webkit.org/272455@main">https://commits.webkit.org/272455@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9ffe9195e6d8264b17fa6262fb19ac1c7f5ccaa6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31705 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10388 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34198 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/28710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32490 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12745 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7636 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28302 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32058 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/8755 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/28297 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/7552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/7715 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/28211 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35543 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/28819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/28658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/33835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7809 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/5814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/31690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/9464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/28017 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8488 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4140 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8325 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cookiecrook commented on LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.html:
```diff
@@ -129,6 +137,14 @@ <h3 data-expectedlabel="alt-before label alt-after" data-testname="heading name
+<h1><a href="https://drafts.csswg.org/css-content/#alt">Complex Alternative Text for  CSS content (previously `alt:`)</a> in pseudo-elements</h1>
+<p>rendered text should be "before label after"</p>
+<p>accessibility label should be "start alt-before end label start alt-after end"</p>
+<button data-alt-text-before="alt-before" data-alt-text-after="alt-after" data-expectedlabel="start alt-before end label start alt-after end" data-testname="button name from complex fallback content with ::before and ::after" class="ex fallback-before-complex fallback-after-complex">label</button><br>
```
We may add some other “complex”-ities to the tests in the future, so please make the test name include `attr()` to describe what makes it complex. It will help avoid name conflict in future test additions.

nt1m committed:
[css-content] AX: Add support for alt text syntax
https://bugs.webkit.org/show_bug.cgi?id=159022
rdar://26942023

Reviewed by Darin Adler.

This adds support for the `content: "foo" / "alt-text";` syntax. Multiple strings / attr() functions can also be chained.

Counters are not supported by this PR. Other non-text types are intentionally not supported as alt-text per-spec.

Spec: https://drafts.csswg.org/css-content-3/#alt

* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content.html:
* LayoutTests/imported/w3c/web-platform-tests/accname/name/comp_name_from_content-expected.txt:
* Source/WebCore/css/CSSValuePair.cpp:
(WebCore::CSSValuePair::CSSValuePair):
(WebCore::CSSValuePair::create):
(WebCore::CSSValuePair::createSlashSeparated):
(WebCore::CSSValuePair::createNoncoalescing):
* Source/WebCore/css/CSSValuePair.h:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeContent):
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueContent):

Canonical link: https://commits.webkit.org/272455@main
